keras/layers/core.py

```diff
@@ -75,79 +119,40 @@ def get_params(self):
+        super(Merge,self).__init__(name, prev, input_dim)
         self.mode = mode
-        self.models = models
-        self.params = []
-        self.regularizers = []
-        self.constraints = []
-        for m in self.models:
-            for i in range(len(m.params)):
-                if not m.params[i] in self.params:
-                    self.params.append(m.params[i])
-                    self.regularizers.append(m.regularizers[i])
-                    self.constraints.append(m.constraints[i])
-
-    def get_params(self):
-        return self.params, self.regularizers, self.constraints
 
-    def get_output(self, train=False):
+    def get_output(self, train):
+        inputs = self.get_input(train)
         if self.mode == 'sum':
-            s = self.models[0].get_output(train)
-            for i in range(1, len(self.models)):
-                s += self.models[i].get_output(train)
+            s = inputs[0]
+            for inp in inputs[1:]:
+                s += inp
             return s
         elif self.mode == 'concat':
-            inputs = [self.models[i].get_output(train) for i in range(len(self.models))]
             return T.concatenate(inputs, axis=-1)
         else:
             raise Exception('Unknown merge mode')
 
-    def get_input(self, train=False):
-        res = []
-        for i in range(len(self.models)):
-            o = self.models[i].get_input(train)
-            if not type(o) == list:
-                o = [o]
-            for output in o:
-                if output not in res:
-                    res.append(output)
-        return res
-
-    @property
-    def input(self):
-        return self.get_input()
-
-    def get_weights(self):
-        weights = []
-        for m in self.models:
-            weights += m.get_weights()
-        return weights
-
-    def set_weights(self, weights):
-        for i in range(len(self.models)):
-            nb_param = len(self.models[i].params)
-            self.models[i].set_weights(weights[:nb_param])
-            weights = weights[nb_param:]
-
-    def get_config(self):
-        return {"name":self.__class__.__name__,
-            "models":[m.get_config() for m in self.models],
-            "mode":self.mode}
-
+    def get_output_dim(self, input_dims):
+        if None in input_dims:
+            print('One or more input dimensions to this layer are unknown. Either fix that or provide the input dimensions manually')
+        if self.mode == 'sum':
+            return tuple(input_dims[0])
+        if self.mode == 'concat':
+            sizes = [input_dim[-1] for input_dim in input_dims]
+            output_dim = list(input_dims[0])  # make a mutable copy
+            output_dim[-1] = sum(sizes)
+            return tuple(output_dim)
 
 class Dropout(Layer):
     '''
         Hinton's dropout.
     '''
-    def __init__(self, p):
-        super(Dropout,self).__init__()
+    def __init__(self, p, name=None, prev=None, input_dim=(None, )):
```

Comment:
I think this setup brings a lot of code redundancy. Why not call `super` with the keyword arguments to `__init__`?